raiden/tests/utils/smoketest.py

```diff
@@ -311,10 +321,19 @@ def setup_raiden(
         registry_address=contract_addresses[CONTRACT_TOKEN_NETWORK_REGISTRY],
         contract_manager=contract_manager,
     )
-    registry.add_token(
-        token_address=to_canonical_address(token.contract.address),
-        given_block_identifier='latest',
-    )
+
+    if contracts_version == DEVELOPMENT_CONTRACT_VERSION:
+        registry.add_token_with_limits(
+            token_address=to_canonical_address(token.contract.address),
+            channel_participant_deposit_limit=RED_EYES_PER_CHANNEL_PARTICIPANT_LIMIT,
+            token_network_deposit_limit=RED_EYES_PER_TOKEN_NETWORK_LIMIT,
+            given_block_identifier='latest',
+        )
+    else:
+        registry.add_token_without_limits(
+            token_address=to_canonical_address(token.contract.address),
+            given_block_identifier='latest',
```

Comment:
Note: The places which are using `given_block_identifier='latest'` will have to be upgraded to follow the consistent view

Reply:
I believe we use `latest` only in tests... i have checked if i am using this in other places in the code but my code doesn't really change anything that wasn't already using "latest".

Reply:
Well, but the problem persists, the tests are susceptible to the same race conditions, I would argue that in the tests because we use Poa it's even more likely to happen.